File: find_pi.cs
using System;
public class Hello
{
    public static void Main()
    {
        double pi = Math.PI;
        double result = 0;
        int place = 0;
        bool limit = true;

        while(limit)
        {
            Console.Write("Enter how many decimal places of Pi to be displayed [LIMIT 15]: ");
            place = int.Parse(Console.ReadLine());
            try
            {
                result = Math.Round(pi, place);
                limit = false;
            }
            catch(ArgumentOutOfRangeException e)
            {
                Console.WriteLine("{0} Please enter a valid value.", e);
                limit = true;
            }
        }      
        Console.Write("Pi in " + place + " decimal places is: " + result);
    }
}

Hello.Main();

File: find_pi.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>
#include <float.h>

int main(void)
{
    double pi = 4.0 * atan(1.0);
    int place;
    bool limit = true;

    while (limit)
    {
        printf("Enter how many decimal places of Pi to be displayed [LIMIT 15]: ");
        scanf("%i", &place);

        if (place < 16 && place >= 0)
            limit = false;
        else
        {
            printf("Decimal place out of bounds! Please enter a valid value.\n");
        }
    }
    printf("Pi in %i decimal places is: %.*f", place, place, pi);
}